### PR TITLE
[Tests-Only] Add test for hiding search bar config option

### LINKED ISSUE #3818: UI test for "Hide search bar" setting
a test is needed for https://github.com/owncloud/phoenix/pull/3817/
the main challenge here will be to edit the config json file from the test step, and maybe a reload of phoenix is needed alsoe

alternatively we can have two separate tests for OCIS & for oC10, with separate config json files as input

### PR DESCRIPTION
**Pull request by dpakach**
## Description

Add UI tests for hiding search bar option

## Related Issue

- Fixes #3818

## How Has This Been Tested?

:robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: tests/acceptance/features/webUIFiles/options.feature [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    Then the search bar should be visible in the webUI
+
+  Scenario: Set config to hide the search bar while the user is logged in
+    Given the property "hideSearchBar" of "options" has been set to false in phoenix config file
+    And user "user1" has been created with default attributes
+    And user "user1" has logged in using the webUI
+    When the user has browsed to the files page
+    Then the search bar should be visible in the webUI
+    When the property "hideSearchBar" of "options" is changed to true in phoenix config file
+    When the user browses to the files page
```
Comment (kiranparajuli589) [inline tests/acceptance/features/webUIFiles/options.feature:24]:
```suggestion
    And the user browses to the files page
```

File: tests/acceptance/features/webUIFiles/options.feature [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    Then the search bar should not be visible in the webUI
+
+  Scenario: Set config to show the search bar while the user is logged in
+    Given the property "hideSearchBar" of "options" has been set to true in phoenix config file
+    And user "user1" has been created with default attributes
+    And user "user1" has logged in using the webUI
+    When the user has browsed to the files page
+    Then the search bar should not be visible in the webUI
+    When the property "hideSearchBar" of "options" is changed to false in phoenix config file
+    When the user browses to the files page
```
Comment (kiranparajuli589) [inline tests/acceptance/features/webUIFiles/options.feature:34]:
```suggestion
    And the user browses to the files page
```

File: tests/acceptance/features/webUIFiles/options.feature [hunk 3]
```diff
@@ -0,0 +1,35 @@
+Feature: Set config options from phoenix config file
+
+  Scenario: Set config to hide the search bar
+    Given the property "hideSearchBar" of "options" has been set to true in phoenix config file
+    And user "user1" has been created with default attributes
+    And user "user1" has logged in using the webUI
```
Comment (jasson99) [inline tests/acceptance/features/webUIFiles/options.feature:6]: Can this be written in background?

File: tests/acceptance/features/webUIFiles/options.feature [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    When the user has browsed to the files page
```
Comment (jasson99) [inline tests/acceptance/features/webUIFiles/options.feature:7]:
```suggestion
    When the user browses to the files page
```

File: tests/acceptance/features/webUIFiles/options.feature [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    Then the search bar should not be visible in the webUI
+
+  Scenario: Set config to show the search bar
+    Given the property "hideSearchBar" of "options" has been set to false in phoenix config file
+    And user "user1" has been created with default attributes
+    And user "user1" has logged in using the webUI
+    When the user has browsed to the files page
```
Comment (jasson99) [inline tests/acceptance/features/webUIFiles/options.feature:14]:
```suggestion
    When the user browses to the files page
```

File: tests/acceptance/stepDefinitions/generalContext.js [hunk 6]
```diff
@@ -34,6 +34,43 @@ Given('the property {string} has been set to {string} in phoenix config file', f
...
+) {
+  return setconfig(key, subkey, false, this.fullPathOfConfigFile)
+})
+
+When('the property {string} of {string} is changed to true in phoenix config file', function(
+  subkey,
+  key
+) {
+  return setconfig(key, subkey, true, this.fullPathOfConfigFile)
+})
```
Comment (kiranparajuli589) [inline tests/acceptance/stepDefinitions/generalContext.js:65]:
```suggestion
When('the property {string} of {string} is changed to {string} in phoenix config file', function(
  subkey,
  key,
  value
) {
  return setconfig(key, subkey, value, this.fullPathOfConfigFile)
})
```

File: tests/acceptance/stepDefinitions/generalContext.js [hunk 7]
```diff
@@ -34,6 +34,43 @@ Given('the property {string} has been set to {string} in phoenix config file', f
...
+
+When('the property {string} of {string} is changed to false in phoenix config file', function(
+  subkey,
+  key
+) {
+  return setconfig(key, subkey, false, this.fullPathOfConfigFile)
+})
+
```
Comment (kiranparajuli589) [inline tests/acceptance/stepDefinitions/generalContext.js:73]: not needed!

File: tests/acceptance/features/webUIFiles/options.feature [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    Then the search bar should be visible in the webUI
+    When the property "hideSearchBar" of "options" is changed to true in phoenix config file
+    When the user browses to the files page
+    Then the search bar should not be visible in the webUI
+
+  Scenario: Set config to show the search bar while the user is logged in
+    Given the property "hideSearchBar" of "options" has been set to true in phoenix config file
+    And user "user1" has been created with default attributes
+    And user "user1" has logged in using the webUI
+    When the user has browsed to the files page
```
Comment (jasson99) [inline tests/acceptance/features/webUIFiles/options.feature:31]:
```suggestion
    When the user browses to the files page
```

File: tests/acceptance/stepDefinitions/generalContext.js (hunk 6 above)
Comment (dpakach) [reply tests/acceptance/stepDefinitions/generalContext.js:65]: If we do that then we will also need to send the type of the value and that IMO would not look very nice

File: tests/acceptance/features/webUIFiles/options.feature [hunk 9]
```diff
@@ -0,0 +1,34 @@
+Feature: Set config options from phoenix config file
+
+  Background:
+    And user "user1" has been created with default attributes
+
+  Scenario: Set config to hide the search bar
+    Given the property "hideSearchBar" of "options" has been set to true in phoenix config file
+    And user "user1" has logged in using the webUI
```
Comment (kiranparajuli589) [inline tests/acceptance/features/webUIFiles/options.feature:8]: This can go to backgrounds section too...

Comment (dpakach) [reply tests/acceptance/features/webUIFiles/options.feature:8]: I just want to check both cases where config is changed before logging in and also when the user logged in

**Pull request merged**